runtime/dancebox/src/lib.rs

```diff
@@ -849,9 +953,36 @@ impl RegistrarHooks for DanceboxRegistrarHooks {
                 e,
             );
         }
+        // Remove bootnodes from pallet_data_preservers
+        DataPreservers::para_deregistered(para_id);
 
         Weight::default()
     }
+
+    fn check_valid_for_collating(para_id: ParaId) -> DispatchResult {
+        // To be able to call mark_valid_for_collating, a container chain must have bootnodes
+        if DataPreservers::boot_nodes(para_id).len() > 0 {
```

Comment:
You can probably add this as a trait that is implemented for the data-preservers palet, and you pass this as config in the registrar pallet. That way you can return an error of the pallet itself